src/sentry/utils/kvstore/bigtable.py

```diff
@@ -0,0 +1,287 @@
+
+        self.project = project
+        self.instance = instance
+        self.table_name = table_name
+        self.client_options = client_options
+        self.default_ttl = default_ttl
+        self.compression = compression
+
+        self.__table: Table
+        self.__table_lock = Lock()
+
+    def _get_table(self, admin: bool = False) -> Table:
+        if admin is True:
+            return (
+                bigtable.Client(project=self.project, admin=True, **self.client_options)
+                .instance(self.instance)
+                .table(self.table_name)
+            )
+
+        try:
+            # Fast check for an existing table
+            return self.__table
+        except AttributeError:
+            # If missing, we acquire our lock to initialize a new one
+            with self.__table_lock:
+                # It's possible that the lock was blocked waiting on someone
+                # else who already initialized, so we first check again to make
+                # sure this isn't the case.
+                try:
+                    table = self.__table
+                except AttributeError:
+                    table = self.__table = (
+                        bigtable.Client(project=self.project, **self.client_options)
+                        .instance(self.instance)
+                        .table(self.table_name)
+                    )
+            return table
+
+    def get(self, key: str) -> Optional[bytes]:
+        row = self._get_table().read_row(key)
+        if row is None:
+            return None
+
+        return self.__decode_row(row)
+
+    def get_many(self, keys: Sequence[str]) -> Iterator[Tuple[str, bytes]]:
+        rows = RowSet()
+        for key in keys:
+            rows.add_row_key(key)
+
+        for row in self._get_table().read_rows(row_set=rows):
+            value = self.__decode_row(row)
+
+            # Even though Bigtable in't going to return empty rows, an empty
+            # value may be returned by ``__decode_row`` if the the row has
+            # outlived its TTL, so we need to check its value here.
+            if value is not None:
+                yield row.row_key.decode("utf-8"), value
+
+    def __decode_row(self, row: PartialRowData) -> Optional[bytes]:
+        columns = row.cells[self.column_family]
+
+        try:
+            cell = columns[self.data_column][0]
+        except KeyError:
+            logger.warning("Retrieved row (%r) which does not contain a data column!", row.row_key)
+            return None
+
+        # Check if a TTL column exists for this row. If there is, we can use
+        # the `timestamp` property of the cells to see if we should return the
+        # row or not.
+        if self.ttl_column in columns:
+            # NOTE: If you need the value, it can be unpacked with the struct.
+            if cell.timestamp < timezone.now():
+                return None
+
+        value = cast(bytes, cell.value)
```

Comment:
Could also be an `assert` for the sake of typing/testing, not strong opinion one way or another.